spring-cloud-gcp-pubsub-stream-binder/src/main/java/org/springframework/cloud/gcp/stream/binder/pubsub/provisioning/PubSubChannelProvisioner.java

```diff
@@ -54,8 +58,10 @@ public ProducerDestination provisionProducerDestination(String name,
 	public ConsumerDestination provisionConsumerDestination(String name, String group,
 			ExtendedConsumerProperties<PubSubConsumerProperties> properties)
 			throws ProvisioningException {
+		// Use group name as subscription name
+		// Generate anonymous random group, if one not provided
```

Comment:
@artembilan The concern I have here is that there doesn't seem to be anything in the binder to clean the subscriptions up when the application shuts down. Should we put in some mechanism for that?

Reply:
I think so. When consumer with such an anonymous group is shouting down we definitely should remove its subscription.

Good catch!

Let's address it in the separate issue though!?